include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -129,6 +130,62 @@ namespace realization {
                  */ 
                 this->Simulation_Time_Object = std::make_shared<Simulation_Time>(simulation_time_config);
 
+                /**
+                 * Read the layer descriptions
+                */
+
+                // try to get the json node
+                auto layers_json_array = tree.get_child_optional("layers");
+
+                // check to see if the node existed
+                if (!layers_json_array) {
+                    // layer description struct
+                        ngen::LayerDescription layer_desc;
+
+                        // extract and store layer data from the json
+                        layer_desc.name = "surface layer";
+                        layer_desc.id = 0;
+                        layer_desc.time_step = 3600;
+                        layer_desc.time_step_units = "s";
+
+                        // add the layer to storage
+                        layer_storage.put_layer(layer_desc, layer_desc.id);
+                }
+                else
+                {
+                    for (std::pair<std::string, boost::property_tree::ptree> layer_config : *layers_json_array) 
+                    {
+                        // layer description struct
+                        ngen::LayerDescription layer_desc;
+
+                        // extract and store layer data from the json
+                        layer_desc.name = layer_config.second.get<std::string>("name");
+                        layer_desc.id = layer_config.second.get<int>("id");
+                        layer_desc.time_step = layer_config.second.get<double>("time_step");
+                        boost::optional<std::string> layer_units = layer_config.second.get_optional<std::string>("time_step_units");
+                        if (*layer_units == "") layer_units = "s";
+                        layer_desc.time_step_units = *layer_units;
+
+                        // check to see if this layer was allready defined
+                        if (layer_storage.exists(layer_desc.id) )
```

Comment:
Why isn't this check inside `put_layer`?

Reply:
because it was a void function that does not throw

Reply:
also you might want to place a new layer at some id, I this case we don't but there is the possibility that layers would be removed in some sort of transition for a highres pass.